com.unity.netcode.gameobjects/Runtime/Transports/UTP/BatchedSendQueue.cs

```diff
@@ -43,18 +50,32 @@ private int TailIndex
         }
 
         public int Length => TailIndex - HeadIndex;
-
         public bool IsEmpty => HeadIndex == TailIndex;
-
         public bool IsCreated => m_Data.IsCreated;
 
         /// <summary>Construct a new empty send queue.</summary>
         /// <param name="capacity">Maximum capacity of the send queue.</param>
         public BatchedSendQueue(int capacity)
         {
-            m_Data = new NativeArray<byte>(capacity, Allocator.Persistent);
+            // Make sure the maximum capacity will be even.
+            m_MaximumCapacity = capacity + (capacity & 1);
+
+            // We pick the minimum capacity such that if we keep doubling it, we'll eventually hit
+            // the maximum capacity exactly. The alternative would be to use capacities that are
+            // powers of 2, but this can lead to over-allocating quite a bit of memory (especially
+            // since we expect maximum capacities to be in the megabytes range). The approach taken
+            // here avoids this issue, at the cost of not having allocations of nice round sizes.
+            m_MinimumCapacity = m_MaximumCapacity;
+            while (m_MinimumCapacity / 2 >= MinimumMinimumCapacity)
```

Comment:
Just curious why you are using /2 as opposed to >> 1 here.
```
while (m_MinimumCapacity >= MinimumMinimumCapacity)
{
    m_MinimumCapacity = m_MinimumCapacity >> 1;
}
```
Wouldn't the above yield the same result?
_(or perhaps I am missing something else here)_

Reply:
That would indeed yield the same result. I chose to use a division operation because I thought it better reflected the intention of the code, which is to halve the capacity until halving it further would drop it below the minimum. Dividing by two makes the code easier to understand (in my opinion).

Regarding any performance cost of using divisions instead of bit shifts, when dividing by constants like that I'd fully expect the compiler/runtime to make the optimization on its own. [This snippet](https://godbolt.org/z/jo7zE7MK8) indicates that this is what happens (at least after JIT-compilation).

Reply:
Right on.
I was thinking code clarity... and you are always safe from divide by zero since the right hand operand is always 2 (which is when you would want to use shift right vs divide).
Thanks for the additional info...excited to get this update!  :)